specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2022-12-01-preview/elasticsan.json

```diff
@@ -0,0 +1,2341 @@
+            "name": "Action",
+            "modelAsString": false
+          },
+          "default": "Allow",
+          "description": "The action of virtual network rule."
+        },
+        "state": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "provisioning",
+            "deprovisioning",
+            "succeeded",
+            "failed",
+            "networkSourceDeleted"
+          ],
+          "x-ms-enum": {
+            "name": "State",
+            "modelAsString": false
+          },
+          "description": "Gets the state of virtual network rule."
+        }
+      },
+      "required": [
+        "id"
+      ],
+      "description": "Virtual Network rule."
+    },
+    "PrivateEndpointConnection": {
+      "type": "object",
+      "description": " Response for PrivateEndpoint Connection object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Private Endpoint Connection Properties.",
+          "$ref": "#/definitions/PrivateEndpointConnectionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "PrivateEndpointConnectionProperties": {
+      "type": "object",
+      "description": " Response for PrivateEndpoint connection properties",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning State of Private Endpoint connection resource",
+          "$ref": "#/definitions/ProvisioningState",
+          "readOnly": true
+        },
+        "privateEndpoint": {
+          "description": "Private Endpoint resource",
+          "$ref": "#/definitions/PrivateEndpoint"
+        },
+        "privateLinkServiceConnectionState": {
+          "description": "Private Link Service Connection State.",
+          "$ref": "#/definitions/PrivateLinkServiceConnectionState"
+        },
+        "groupIds": {
+          "description": " List of resources private endpoint is mapped",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "privateLinkServiceConnectionState"
+      ]
+    },
+    "PrivateEndpoint": {
+      "type": "object",
+      "description": "Response for PrivateEndpoint",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ARM identifier for Private Endpoint",
+          "readOnly": true
+        }
+      }
+    },
+    "PrivateLinkServiceConnectionState": {
+      "type": "object",
+      "description": "Response for Private Link Service Connection state",
+      "properties": {
+        "status": {
+          "description": "Indicates whether the connection has been Approved/Rejected/Removed by the owner of the service.",
+          "$ref": "#/definitions/PrivateEndpointServiceConnectionStatus"
```

Comment:
>PrivateEndpointServiceConnectionStatus [](http://example.com/codeflow?start=33&length=38)

You should use the TrackedResource definition from v5 common types.

Reply:
Keeping it to v3, due to hard deadline, in GA swagger will keep this in mind and move to v5. Thanks